ac.soton.xeventb.xmachine/xtend-gen/ac/soton/xeventb/xmachine/validation/XMachineValidator.java

```diff
@@ -309,36 +310,87 @@ private EStructuralFeature getFeature(final EObject obj, final IAttributeType at
   private void createIssue(final EObject obj, final IMarker rodinMarker) {
     try {
       final IAttributeType attributeType = RodinMarkerUtil.getAttributeType(rodinMarker);
-      final EStructuralFeature feature = this.getFeature(obj, attributeType);
       final Map<String, Object> attributes = rodinMarker.getAttributes();
       final Object severity = attributes.get(IMarker.SEVERITY);
-      final int start = RodinMarkerUtil.getCharStart(rodinMarker);
-      final int end = RodinMarkerUtil.getCharEnd(rodinMarker);
       Object message = attributes.get(IMarker.MESSAGE);
-      if ((start != (-1))) {
-        String _plus = (message + " (from ");
-        String _plus_1 = (_plus + Integer.valueOf(start));
-        String _plus_2 = (_plus_1 + " to ");
-        String _plus_3 = (_plus_2 + Integer.valueOf(end));
-        String _plus_4 = (_plus_3 + ")");
-        message = _plus_4;
+      EStructuralFeature feature = this.getFeature(obj, attributeType);
+      EObject elem = obj;
+      if ((feature == null)) {
+        elem = this.getMachine(obj);
```

Comment:
could you use 
((EventBObject)obj).getContaining(MachinePackage.Literals.MACHINE);